ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -473,7 +483,7 @@ private CloudStorageException toCloudStorageException(String message, Exception
       statusCode = StatusCodes.INTERNAL_SERVER_ERROR;
     }
     // Everything is retryable except NOT_FOUND
-    boolean isRetryable = (statusCode != StatusCodes.NOTFOUND);
+    boolean isRetryable = (statusCode != StatusCodes.NOTFOUND && !(e instanceof StoreException));
```

Comment:
What is the case where a `StoreException` gets here? Is this when the CloudUpdateValidator throws?

Reply:
yes.